Assets/MixedRealityToolkit.Services/InputSystem/MixedRealityInputSystem.cs

```diff
@@ -390,7 +390,7 @@ public override void HandleEvent<T>(BaseEventData eventData, ExecuteEvents.Event
                 return;
             }
 
-            Debug.Assert(baseInputEventData.InputSource.Pointers != null, $"InputSource {baseInputEventData.InputSource.SourceName} doesn't have any registered pointers! Input Sources without pointers should use the GazeProvider's pointer as a default fallback.");
+            if (baseInputEventData.InputSource.Pointers == null) { Debug.LogError($"InputSource {baseInputEventData.InputSource.SourceName} doesn't have any registered pointers! Input Sources without pointers should use the GazeProvider's pointer as a default fallback."); }
```

Comment:
Debug.Assert gets compiled out in many cases. With this change, now we have an empty `if null` check in those cases, instead of just nothing. That feels weird, especially for something that won't exist when the app is actually built to be released.

Reply:
Yep - if you're profiling on anything other than the master build (on device) you will see allocations caused by string manipulations.

Reply:
The impact in the editor is really bad, though. We can't count on devs profiling on device to iron out performance, so they'll see the core input system generating tons of garbage and not necessarily understand that it won't follow them onto the device.

An alternative is to pass an unformatted message to the assert.